mesh/nurbs.cpp

```diff
@@ -60,6 +107,165 @@ KnotVector &KnotVector::operator=(const KnotVector &kv)
    return *this;
 }
 
+int KnotVector::GetSpan(real_t u) const
+{
+   int low, mid, high;
+
+   if (u == knot(NumOfControlPoints+Order))
+   {
+      mid = NumOfControlPoints-1;
+   }
+   else if (u == knot(0))
+   {
+      mid = Order;
+   }
+   else if ((u > knot(0)) && (u < knot(NumOfControlPoints+Order)))
+   {
+      low = Order;
+      high = NumOfControlPoints;
+      mid = (low + high)/2;
+      while ( (u < knot(mid)) || (u >= knot(mid+1)) )
+      {
+         if (u < knot(mid))
+         {
+            high = mid;
+         }
+         else
+         {
+            low = mid;
+         }
+         mid = (low + high)/2;
+      }
+   }
+   else
+   {
+      mfem_error("Knot location outside of the range of the KnotVector");
+   }
+
+   return mid;
+}
+
+real_t KnotVector::GetGreville(int i) const
+{
+   real_t sum = 0.0;
+   for (int j = 1; j < Order+1; j++) { sum += knot[i + j]; }
+   return sum/real_t(Order);
+}
+
+real_t KnotVector::GetBotella(int i) const
```

Comment:
Do you think it might be worth it to cache the results of `GetBotella`? I noticed `KnotVector` now stores Demko points ([nurbs.hpp](https://github.com/mfem/mfem/pull/4326/files#diff-c75875c788ff181d814edcb659958b967abf8742fcfcc35e64a10766d87f397cR57)) but throughout the code it seems like Botella points are used quite frequently (maybe more frequently than Demko points?)